uaclient/entitlements/esm.py

```diff
@@ -1,13 +1,22 @@
 from uaclient.entitlements import repo
 
 
-class ESMInfraEntitlement(repo.RepoEntitlement):
-
+class ESMBaseEntitlement(repo.RepoEntitlement):
     help_doc_url = "https://ubuntu.com/esm"
+    origin = "UbuntuESM"
+    repo_pin_priority = "never"
+    disable_apt_auth_only = True  # Only remove apt auth files when disabling
```

Comment:
I think now we need to change this to a @property class method that looks at our util.get_platform_info()['series'] because on trusty is the only place we want to continue advertising esm apt sources right?

Reply:
Great point, let me look at that.

Reply:
Also pushed the following fixes:

- unset ESM*.repo_pin_priorityand
- fixed UbuntuESM vs UbuntuESMApps origin
- more strict aptURL matching in apt-policy